### ISSUE TITLE
Can't change to another SMS application even after uninstalling

### ISSUE THREAD

**Issue by Grah96** [SilenceIM/Silence#394]
Two serious problems:
1. I try to use another text message application, change the default SMS application and... nothing - I can get no text messages and cannot send any. It's like Silence has silenced alternative SMS apps! I uninstall Silence... still I get nothing. Can't send or receive texts.  I reinstall Silence (or SMS Secure as it was then) and it starts working again. It is like it is locked onto Silence's settings. 
2. What I have also realised all my backup programs are doing nothing but pointlessly backing up the text messages I had before I installed Silence/SMS Secure. I can export plain text file containing the Silence stored text messages but then what? Do I replace the existing stock stored SMS file? Excuse my ignorance but how exactly  As I think others have mentioned this needs to be a lot clearer and transparent please, and there probably should be some warning before installing.

This is extremely frustrating. Please advise how I can fix both of these problems.
thanks
Graham

Sony Xperia Z3 Android 5.1.1

**Comment by Grah96** [comment]
https://gist.github.com/25b211d7ed9c797b7c9a1bc7a waa99497b2

**Comment by Grah96** [comment]
The above file is the log file for analysis. Still waiting for some acknowledgement of the problem.

**Comment by pR0Ps** [comment]
Your log file URL is malformed.

To change your default SMS application go to Settings -> Wireless & networks -> More -> Default SMS app. Tap it and a menu will appear allowing you to choose which SMS application to use. When you uninstall Silence you may have to change this option manually but it should (as with uninstalling any other SMS app) just default back to another app when the default app is uninstalled.

There is no way for any (non-root) application to "block" SMS messages from being received after it's been uninstalled. If what you say is true, it's a bug in the custom flavor of Android that Sony is distributing with your phone and not Silence.

Silence stores it's SMS messages in a custom database that can be password protected so no other rouge apps can access the messages. When exporting a plaintext backup, it says in the export dialog "Compatible with 'SMS Backup & Restore'". If you install [SMS Backup & Restore](https://play.google.com/store/apps/details?id=com.riteshsahu.SMSBackupRestore), you can point it at the file Silence exports and it will import all the messages into the system SMS database.

#145 (forwarding messages into the system database) is something that we've discussed as a possible solution for people who don't care if their messages are readable in the system SMS database and want SMS backup tools to work properly with Silence.

**Comment by ArchangeGabriel** [comment]
Note that currently, there is no way to export MMS out of Silence for reimporting anywhere else (only thing working is Silence backup).

**Comment by Grah96** [comment]
Thanks very much for the comments. I've still had no luck. I exported plain text messages from Silence,installed the backup program suggested and attempted to import. It appeared to work but, when I switch to the stock sms app, still only has messages until January, which is when I installed SMS Secure. I disabled and reenabled stock sms app, which the sms backup and restore app said might be necessary. Rebooted phone etc. No post January messages appear. The location of the stock sms database does not appear to be visible so I can't manually copy the file. What is more I still cannot send or receive sms messages from any other sms app apart from Silence. (The stock sms app just says "cannot send"). If it's a Sony issue 'm wondering if a hard reset is the only solution. 
Being able to forward messages to the system database would be a useful option to have.
